src/main/java/org/veupathdb/service/eda/ss/model/reducer/BinaryValuesStreamer.java

```diff
@@ -134,24 +136,26 @@ public Iterator<VariableValueIdPair<List<Long>>> streamAncestorIds(Entity descen
    */
   public Iterator<VariableValueIdPair<List<String>>> streamIdMap(Entity entity, Study study) throws IOException {
     Path path = binaryFilesManager.getIdMapFile(study, entity, BinaryFilesManager.Operation.READ);
-    final ListConverter<String> listConverter = new ListConverter<>(
-        new StringValueConverter(BYTES_RESERVED_FOR_ID),
-        entity.getAncestorEntities().size() + 1); // First entry of list is for entity ID, rest are ancestor IDs.
-    final ValueWithIdDeserializer<List<String>> ancestorsWithId = new ValueWithIdDeserializer<>(listConverter);
+    List<Integer> bytesReservedForAncestors = binaryFilesManager.getBytesReservedForAncestry(study, entity);
+    Integer bytesReservedForId = binaryFilesManager.getBytesReservedForEntity(study, entity);
+    IdsMapConverter converter = new IdsMapConverter(bytesReservedForAncestors, bytesReservedForId);
+
     return new FilteredValueIterator<>(path,
         x -> true, // Do not apply any filters.
-        ancestorsWithId,
+        converter,
         Function.identity());
   }
 
   public Iterator<Long> streamUnfilteredEntityIdIndexes(Study study, Entity entity) throws IOException {
-    ListConverter<String> converter = new ListConverter<>(new StringValueConverter(BYTES_RESERVED_FOR_ID), entity.getAncestorEntities().size() + 1);
+    List<Integer> bytesReservedForAncestors = binaryFilesManager.getBytesReservedForAncestry(study, entity);
```

Comment:
perhaps factor this reused code into getConverter(study, entity)?

Reply:
Sounds good, added this.